### PR TITLE
stopcovid russia

### PR DESCRIPTION
**Pull request by abhidg**
- parsers: add stopcovid-russia
- update
- update

### CODE REVIEW

File: isaric/parsers/stopcovid-russia.toml [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+  2 = "> 2-4 weeks"
+  3 = ">1-2 months"
+  4 = ">2-3 months"
+  5 = ">3-6 months"
+  7 = "Since the hospital discharge"
+
+[adtl.defs.severity.values]
+  1 = "caused bearable discomfort/was controlled"
+  2 = "caused severe discomfort/was not controlled"
+  3 = "the reason for calling an ambulance/visit a doctor"
```
Comment (pipliggins) [inline isaric/parsers/stopcovid-russia.toml:50]: Both unused, suggest removing

File: isaric/parsers/stopcovid-russia.toml [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+  name = "runny_nose"
+  date.ref = "admissionDate"
+  phase = "admission"
+  is_present = { field = "runnynose_ceoccur_v2", ref = "Y/N" }
+
+[[observation]] # TODO: use combinedType here
+  name = "cough"
+  date.ref = "admissionDate"
+  phase = "admission"
+  is_present = { field = "cough_ceoccur_v2", ref = "Y/N" }
```
Comment (pipliggins) [inline isaric/parsers/stopcovid-russia.toml:437]: Just marking this as a still to-do!

**Review by pipliggins — APPROVED** [review]

File: isaric/parsers/stopcovid-russia.toml [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+  ]
+
+[[observation]]
+  name = "sternal_capillary_refill_time_greater_2s"
+  phase = "admission"
+  date.ref = "admissionDate"
+  is_present = { field = "stercap_vsorres", ref = "Y/N" }
+
+[[observation]]
+  name = "inability_to_walk_scale"
```
Comment (ekamau) [inline isaric/parsers/stopcovid-russia.toml:577]: Just thinking, are the two walking attributes - inability_to_walk and inability_to_walk_scale - related? 

If so, should the inability_to_walk_scale be conditional on the attribute 'inability_to_walk' .. that is, 'inability_to_walk_scale' is valid if the inability_to_walk == yes or true?

**Review by ekamau — APPROVED** [review]
looks good. I had one or two questions in the comments ..

File: isaric/parsers/stopcovid-russia.toml [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+    field = "chroniccard_mhyn"
+    ref = "Y/N"
+
+  [subject.has_chronic_kidney_disease]
+    field = "renal_mhyn"
+    ref = "Y/N"
+
+  [subject.has_diabetes]
+    combinedType = "any"
+    fields = [ # not mapping false, as it does not imply absence of diabetes
```
[the remark this answers is not in the document]
Comment (abhidg) [reply isaric/parsers/stopcovid-russia.toml]: Dropped other fields, so no longer applies

File: isaric/parsers/stopcovid-russia.toml (hunk 3 above)
Comment (abhidg) [reply isaric/parsers/stopcovid-russia.toml:577]: They are related, but usually they are from the same underlying variable. With inability_to_walk == false, inability_to_walk_scale would still have an entry (corresponding to no inability to walk)

**Comment by sadiekelly** [comment]
> @sadiekelly thanks for the review! I have updated the parser -- will wait to hear back from upstream on the HIV fields before merging

@abhidg all looks good to me - thanks!

**Comment by sadiekelly** [comment]
> @sadiekelly thanks for the review! I have updated the parser -- will wait to hear back from upstream on the HIV fields before merging

@abhidg  we have not been able to obtain confirmation from the Russia team regarding the correct coding of the HIV field so suggest not mapping this variable for this parser

**Review by pipliggins — APPROVED** [review]

**Pull request merged**